_includes/specials/bech32/21-brd.md

```diff
@@ -0,0 +1,83 @@
+describes what they learned implementing bech32 and other segwit
+technology for their wallet.*
+
+BRD Wallet began implementing bech32 support in January 2018 with the
+[addition of bech32 decoding and encoding support][brd pr1] to
+[breadwallet-core][], an MIT-licensed cross-platform C library
+with no external dependencies.  All of BRD's software avoids
+third-party library dependencies as much as possible, with it currently
+using only Pieter Wuille's [libsecp256k1][].  Minimizing dependencies is
+typical for a high-security crypto project. For the bech32
+implementation, we found that bech32's [BIP173][] is pretty well
+documented, so there were no specific issues that were complex to
+deal with.
+
+In March 2018, the library was [updated][brd pr2] to automatically parse
+anything provided as a Bitcoin address to determine whether it was a
+legacy P2PKH, legacy P2SH, or segwit bech32 address and to automatically
+generate the appropriate scriptPubKey in each case.  This allowed BRD to
+begin sending to bech32 addresses.  Finally in October 2018, we
+implemented full segwit support across the library backend and mobile
+app frontends, allowing users to begin receiving to bech32 addresses and
+making the default that all change addresses were now bech32.
+
+One thing BRD never implemented was support for receiving to
+P2SH-wrapped segwit addresses, instead going straight to bech32.  This
+was a deliberate design optimization to make the best use of the bloom
+filter mechanism BRD uses to scan for transactions affecting user
+wallets.  To allow users to track when they've been paid, bloom filters
+are matched against each data element in a scriptPubKey.  For a given
+public key, the data element in the scriptPubKey is identical for both
+legacy P2PKH and native segwit (bech32) P2WPKH.  Here's an example
+[previously used][identical spk data] by Optech:
+
+- Legacy P2WPKH scriptPubKey for address 1B6FkNg199ZbPJWG5zjEiDekrCc2P7MVyC:
+
+  <pre>OP_DUP OP_HASH160 OP_PUSH20 <b>6eafa604a503a0bb445ad1f6daa80f162b5605d6</b> OP_EQUALVERIFY OP_CHECKSIG</pre>
+
+- Native segwit (bech32) P2WPKH scriptPubKey for address bc1qd6h6vp99qwstk3z668md42q0zc44vpwkk824zh:
+
+  <pre>OP_0 OP_PUSH20 <b>6eafa604a503a0bb445ad1f6daa80f162b5605d6</b></pre>
+
+Because a bloom filter for a given element will match both P2PKH and
+P2WPKH addresses for the same public key, BRD is able to scan for either
+type of payment with zero additional overhead.  It also makes the
+implementation much cleaner and doesn't increase the resource use of
+public nodes that serve bloom filters.  This may be a worthwhile
+optimization for wallets and services using other types of scanning as
+well, as it may produce less overhead than the separate HD derivation
+path recommended by [BIP84][].
+
+Other observations from BRD's implementation of bech32 support include
+that fee calculation on Bitcoin is hideous---feerates spike multiple
```

Comment:
is this fee estimation in 'Bitcoin' or in 'Bitcoin Core'?

Reply:
I strongly believe he meant Bitcoin in general, as surrounding comments in the interview not included here referred to having a fixed maximum block size.